userland/test/test_publish_subscribe.py.in

```diff
@@ -0,0 +1,28 @@
+from launch import LaunchDescriptor
+from launch.exit_handler import primary_exit_handler
+from launch.launcher import DefaultLauncher
+
+
+def test_publish_subscribe():
+    ld = LaunchDescriptor()
+
+    ld.add_process(
+        cmd=['@TEST_PUBLISHER_EXECUTABLE@'],
+        name='test_publisher',
+    )
+
+    ld.add_process(
+        cmd=['@TEST_SUBSCRIBER_EXECUTABLE@'],
+        name='test_subscriber',
+        exit_handler=primary_exit_handler,
+    )
+
+    launcher = DefaultLauncher()
+    launcher.add_launch_descriptor(ld)
+    rc = launcher.launch()
+
+    assert rc == 0, 'The subsciber did not receive any messages'
+
+
```

Comment:
That's fine, PEP8 is not clear about this issue:

https://www.python.org/dev/peps/pep-0008/#blank-lines

However, single lines are used to separate blocks when they are methods within a class, so not all blocks are separated by more than one line. The PEP8 does allow for use of additional lines (sparingly) to separate unrelated functions, but it doesn't clearly state anything about flow control statements or global variables. I think this case easily falls into the "sparingly" category. I guess my default mode is to use a little vertical whitespace allowed in all cases, but I'm fine with that not being the case if the PEP8 allows it.